src/lox_class.rs

```diff
@@ -15,6 +18,18 @@ pub struct LoxClass {
     pub superclass: Option<Box<LoxClass>>,
 }
 
+impl PartialEq for LoxClass {
+    fn eq(&self, other: &Self) -> bool {
+        std::ptr::eq(self, other)
+    }
+}
+
+impl Hash for LoxClass {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        (self as *const _ as usize).hash(state)
```

Comment:
this is fine as long as the partialeq impl compares type ids, which it does not do right now